xds/type/matcher/v3/domain.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+package xds.type.matcher.v3;
+
+import "xds/annotations/v3/status.proto";
+import "xds/type/matcher/v3/matcher.proto";
+
+import "validate/validate.proto";
+
+option java_package = "com.github.xds.type.matcher.v3";
+option java_outer_classname = "ServerNameMatcherProto";
+option java_multiple_files = true;
+option go_package = "github.com/cncf/xds/go/xds/type/matcher/v3";
+
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Server name matcher]
+
+// Matches a fully qualified server name against a set of wildcard domain
+// names.
+message ServerNameMatcher {
```

Comment:
Upfront, I completely agree we need something here to get parity with current Envoy config, and I have no problem with this being that solution.

Just wondering if we want a more general vs domain specific matcher. It seems like this could be a "PrefixOrExactMatcher" or something generic, where if there is a `*` we treat it as prefix, else exact?

I think the only behavioral implication would be that `*w.example.com` would be valid. I don't have any concerns with that though; its up to the control plane to reject that if it doesn't like it?

Just an idea, I don't have a strong opinion here